File: 이것이 취업을 위한 코딩 테스트다/구현/게임_개발/게임_개발.java
package 구현.게임_개발;

import java.util.Scanner;

public class 게임_개발 {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int rowSize = sc.nextInt();
    int columnSize = sc.nextInt();
    int [][]boards = new int[50][50];
    int [][]checks = new int[50][50];
    int currentRow = sc.nextInt();
    int currentColumn = sc.nextInt();
    int direction = sc.nextInt();
    int []row = {-1, 0, 1, 0};
    int []column = {0, -1, 0, 1};
    int nextRow, nextColumn;
    int answer = 1, count = 0;
    checks[1][1] = 1;

    for (int i = 0; i < rowSize; i++) {
      for (int j = 0; j < columnSize; j++) {
        boards[i][j] = sc.nextInt();
      }
    }

    while (true) {
      direction = direction == 0 ? 3 : direction - 1;
      nextRow = currentRow + row[direction];
      nextColumn = currentColumn + column[direction];

      if (boards[nextRow][nextColumn] == 0 && checks[nextRow][nextColumn] == 0) {
        count = 0;
        answer++;
        checks[nextRow][nextColumn] = 1;
        currentRow = nextRow;
        currentColumn = nextColumn;
      } else {
        count++;
      }

      if (count == 4) {
        nextRow = currentRow - row[direction];
        nextColumn = currentColumn - column[direction];

        if (boards[nextRow][nextColumn] == 0) {
          currentRow = nextRow;
          currentColumn = nextColumn;
          count = 0;
        } else {
          break;
        }
      }
    }

    System.out.println(answer);
  }
}

File: 이것이 취업을 위한 코딩 테스트다/구현/게임_개발/게임_개발.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
  int n, m;
  cin >> n >> m;

  int currentRow, currentColumn, currentDirection;
  cin >> currentRow >> currentColumn >> currentDirection;
  cin.ignore();

  int check[50][50];
  int arr[50][50];
  int temp;
  check[1][1] = 1;
  
  for (int i = 0; i < n; i++) {
    for (int j = 0; j < m; j++) {
      cin >> temp;
      arr[i][j] = temp;
    }
  }

  int row[] = {-1, 0, 1, 0};
  int column[] = {0, 1, 0, -1};
  int nextRow, nextColumn, nextDirection;
  int count = 0;
  int answer = 1;

  while (true) {
    if (count == 4 ) {
      nextRow = currentRow - row[nextDirection];
      nextColumn = currentColumn - column[nextDirection];

      if (arr[nextRow][nextColumn] == 1) {
        break;
      } else {
        currentRow = nextRow;
        currentColumn = nextColumn;
        count = 0;
      }
    }

    nextDirection = currentDirection == 0 ? 3 : currentDirection - 1;
    cout << nextDirection << endl;
    nextRow = currentRow + row[nextDirection];
    nextColumn = currentColumn + column[nextDirection];
    cout << currentRow << " " << currentColumn << endl;

    if (arr[nextRow][nextColumn] == 0 && check[nextRow][nextColumn] == 0) {
      answer++;
      count = 0;
      check[nextRow][nextColumn] = 1;
      currentRow = nextRow;
      currentColumn = nextColumn;
      currentDirection = nextDirection;

      continue;
    } else {
      count++;
      currentDirection = nextDirection;
    }

  }

  cout << answer;
}